src/utils/tokens.ts

```diff
@@ -153,6 +172,353 @@ const tokens = {
+   * @param address Where to send the minted tokens
+   * @param amount Amount of tokens to mint
+   * @param storage Storage instance of the wallet
+   * @param [options={}] Options to mint tokens
+   * @param {string|null} [options.token=null] Token to mint, may be null if we are creating the token
+   * @param {IDataInput|null} [options.mintInput=null] Input to spend, may be null of we are creating the token
+   * @param {boolean} [options.createAnotherMint=true] If a mint authority should be created on the transaction.
+   * @param {boolean} [options.createMelt=false] If a melt authority should be created on the transaction.
+   * @param {string|null} [options.changeAddress=null] The address to send any change output.
+   * @param {boolean} [options.isCreateNFT=false] If this transaction will create an NFT
+   *
+   * @returns {Promise<IDataTx>} The transaction data
+   */
+  async prepareMintTxData(
+    address: string,
+    amount: number,
+    storage: IStorage,
+    {
+      token = null,
+      mintInput = null,
+      createAnotherMint = true,
+      createMelt = false,
+      changeAddress = null,
+      isCreateNFT = false,
+    }: {
+      token?: string|null,
+      mintInput?: IDataInput|null,
+      createAnotherMint?: boolean,
+      createMelt?: boolean,
+      changeAddress?: string|null,
+      isCreateNFT?: boolean,
+    } = {},
+  ): Promise<IDataTx> {
+    const inputs: IDataInput[] = [];
+    const outputs: IDataOutput[] = [];
+    const depositAmount = this.getDepositAmount(amount, isCreateNFT);
+
+    // get HTR deposit inputs
+    let foundAmount = 0;
+    for await (const utxo of storage.selectUtxos({token: HATHOR_TOKEN_CONFIG.uid, target_amount: depositAmount})) {
+      foundAmount += utxo.value;
+      inputs.push({
+        txId: utxo.txId,
+        index: utxo.index,
+        value: utxo.value,
+        address: utxo.address,
+        authorities: utxo.authorities,
+        token: utxo.token,
+      });
+    }
+
+    if (foundAmount < depositAmount) {
+      throw new InsufficientFundsError(`Not enough HTR tokens for deposit: ${depositAmount} required, ${foundAmount} available`);
+    }
+
+    // get output change
+    if (foundAmount > depositAmount) {
+      const cAddress = changeAddress || await storage.getCurrentAddress();
+
+      outputs.push({
+        type: getAddressType(cAddress, storage.config.getNetwork()),
+        address: cAddress,
+        value: foundAmount - depositAmount,
+        timelock: null,
+        token: HATHOR_TOKEN_CONFIG.uid,
+        authorities: 0,
+        isChange: true,
+      });
+    }
+
+    if (mintInput !== null) {
+      // We are spending a mint input to mint more tokens
+      inputs.push(mintInput);
+    }
+
+    // Add output to mint tokens
+    outputs.push({
+      type: 'mint',
+      address,
+      value: amount,
+      timelock: null,
+      authorities: 0,
+    });
+
+    if (createAnotherMint) {
+      const newAddress = await storage.getCurrentAddress();
```

Comment:
We used the old `wallet.getAddressToUse` ([ref](https://github.com/HathorNetwork/hathor-wallet-lib/blob/dev/src/tokens.js#L510)), same with `createAnotherMelt`, so I did the same here.
I also think this should be an optional parameter, but maybe we can change this on another PR since it doesn't change the current behavior.

Reply:
You can just create an issue with this suggestion